port/unix/omrsock.c

```diff
@@ -322,6 +338,37 @@ omrsock_getaddrinfo_protocol(struct OMRPortLibrary *portLibrary, omrsock_addrinf
 	return 0;
 }
 
+int32_t
+omrsock_addrinfo_address(struct OMRPortLibrary *portLibrary, omrsock_addrinfo_t handle, uint32_t index, omrsock_sockaddr_t address)
+{
+	omr_os_addrinfo *info = NULL;
+	omr_os_sockaddr_storage *sockAddr = NULL;
+	uint32_t i = 0;
+
+	if ((NULL == handle) || (NULL == handle->addrInfo) || (index >= handle->length)) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+
+	info = (omr_os_addrinfo *)handle->addrInfo;
+
+	for (i = 0; i < index; i++) {
+		info = info->ai_next;
+		if (NULL == info) {
+			return OMRPORT_ERROR_INVALID_ARGUMENTS;
+		}
+	}
+
+	if(get_os_protocol(info->ai_family) == OS_SOCK_AF_INET6) {
+		sockAddr = (omr_os_sockaddr_storage *)info->ai_addr;
+		memcpy(&address->data, sockAddr, sizeof(omr_os_sockaddr_storage));
```

Comment:
This can be an assignment:
```suggestion
		address->data = *info->ai_addr;
```

Reply:
That's true, it should be an assignment. I will change it.

Reply:
@rwy0717 I realized that address->data is actually not a pointer, so I think we will still need to memcpy the address.
```
/**
 * A struct for storing socket address information. Big enough to store IPv4 and IPv6 addresses.
 */
typedef struct OMRSockAddrStorage {
	omr_os_sockaddr_storage data;
} OMRSockAddrStorage;
```